ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/PostgresDaoSpec.scala

```diff
@@ -170,17 +170,19 @@ class PostgresDaoSpec
       }
     }
 
-    val rejectionReasonGen: Gen[RejectionReason] = for {
-      const <- Gen.oneOf[String => RejectionReason](
-        Seq[String => RejectionReason](
-          RejectionReason.Inconsistent.apply(_),
-          RejectionReason.OutOfQuota.apply(_),
-          RejectionReason.TimedOut.apply(_),
-          RejectionReason.Disputed.apply(_),
-          RejectionReason.DuplicateCommandId.apply(_)
-        ))
-      desc <- Arbitrary.arbitrary[String].filter(_.nonEmpty)
-    } yield const(desc)
+    val rejectionReasonGen: Gen[RejectionReason] =
+      for {
+        const <- Gen.oneOf[String => RejectionReason](
+          Seq[String => RejectionReason](
+            RejectionReason.Inconsistent.apply(_),
+            RejectionReason.OutOfQuota.apply(_),
+            RejectionReason.TimedOut.apply(_),
+            RejectionReason.Disputed.apply(_),
+            RejectionReason.DuplicateCommandId.apply(_)
+          ))
+        // need to use Arbitrary.arbString to get only valid unicode characters
+        desc <- Arbitrary.arbString.arbitrary.filter(_.nonEmpty)
```

Comment:
```suggestion
        desc <- Arbitrary.arbitrary[String].map(_.filterNot(_ == 0)).filter(_.nonEmpty)
```

Reply:
`Arbitrary.arbString.arbitrary` is the same things as `Arbitrary.arbitrary[String]` once you resolve implictis.

The issue is that `Arbitrary.arbString.arbitrary`  is based on `Arbitrary.arbChar` that may generate null character.

Reply:
I actually checked this and it seems to me that `arbString`s are list of of `arbChar`s, where latter is filtered for invalid UTF characters.

See: 
https://github.com/rickynils/scalacheck/blob/master/src/main/scala/org/scalacheck/Arbitrary.scala#L152
and
https://github.com/rickynils/scalacheck/blob/master/src/main/scala/org/scalacheck/Arbitrary.scala#L123

Reply:
`0` is not excluded there however